### PR TITLE
[HOPSWORKS-825] Remove member from project and jobs review

### PR DESCRIPTION
**Pull request by robzor92**
## Make sure there is no duplicate PR for this issue

* **Please check if the PR meets the following requirements**
- [x] Adds tests for the submitted changes (for bug fixes & features)
- [ ] Passes the tests
- [x] HOPSWORKS JIRA issue has been opened for this PR
- [x] All commits have been squashed down to a single commit

* **Post a link to the associated JIRA issue**

* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

    - Data owner should be able to remove data scientists

    - Timers for jobs will be cancelled if the user is not a member of the project where the job is created

* **What is the new behavior (if this is a feature change)?**

* **Does this PR introduce a breaking change?** (What changes might users need to make in their application due to this PR?)

* **Other information**:

### CODE REVIEW

**Comment by robzor92** [comment]
https://github.com/logicalclocks/hops-testing/compare/master...robzor92:HOPSWORKS-825?expand=1

**Comment by robzor92** [comment]
https://github.com/logicalclocks/hops-testing/pull/236

**Comment by robzor92** [comment]
Jenkins test this please

**Comment by robzor92** [comment]
@tkakantousis please review.

File: hopsworks-common/src/main/java/io/hops/hopsworks/common/jobs/JobScheduler.java [hunk 1]
```diff
@@ -93,10 +96,16 @@ public void timeout(Timer timer) throws GenericException, JobException {
...
     //Valid job?
     Jobs job = jobFacade.findById((Integer) jobId);
+
+    //The user may have been removed from the project so any timers for scheduled jobs should be removed too
     if (job == null) {
       logger.log(Level.WARNING, "Trying to run a job with non-existing id, canceling timer.");
       timer.cancel();
       return;
+    } else if(projectTeamFacade.findCurrentRole(job.getProject(), job.getCreator()) == null) {
+      logger.log(Level.WARNING, "Trying to run a job created by a user no longer in this project, canceling timer.");
```
Comment (tkakantousis) [inline hopsworks-common/src/main/java/io/hops/hopsworks/common/jobs/JobScheduler.java]: @robzor92 change to `INFO`

File: hopsworks-ear/test/spec/projects_spec.rb [hunk 2]
```diff
@@ -340,6 +340,24 @@
           memb = json_body.detect { |e| e[:user][:email] == new_member }
           expect(memb).to be_nil
         end
+        it "should remove data scientist from project when remove issued by another data owner" do
```
Comment (tkakantousis) [inline hopsworks-ear/test/spec/projects_spec.rb]: @robzor92 Do we have a test that should fail for a data scientist trying to remove a data owner? If not we need to add one

File: hopsworks-api/src/main/java/io/hops/hopsworks/api/project/ProjectMembersService.java [hunk 3]
```diff
@@ -198,26 +198,29 @@ public Response removeMembersByID(@PathParam("email") String email, @Context Sec
...
     }
-    if (userProjectRole.equals(AllowedProjectRoles.DATA_SCIENTIST) && !owner.getEmail().equals(email)) {
+
+    //Data scientist can only remove themselves
+    String reqUserProjectRole = projectTeamFacade.findCurrentRole(project, reqUser.getEmail());
+    if (reqUserProjectRole.equals(AllowedProjectRoles.DATA_SCIENTIST) && !reqUser.getEmail().equals(email)) {
       throw new ProjectException(RESTCodes.ProjectErrorCode.MEMBER_REMOVAL_NOT_ALLOWED, Level.FINE);
     }
+
+    //Not able to remove project own regardless who is trying to remove the member
```
Comment (tkakantousis) [inline hopsworks-api/src/main/java/io/hops/hopsworks/api/project/ProjectMembersService.java]: @robzor92 `owner`

File: hopsworks-api/src/main/java/io/hops/hopsworks/api/project/ProjectMembersService.java [hunk 4]
```diff
@@ -198,26 +198,29 @@ public Response removeMembersByID(@PathParam("email") String email, @Context Sec
...
     }
-    //Data Scientists are only allowed to remove themselves
+    //Not able to remove members that do not exist
     String userProjectRole = projectTeamFacade.findCurrentRole(project, email);
     if (userProjectRole == null || userProjectRole.isEmpty()) {
       throw new ProjectException(RESTCodes.ProjectErrorCode.TEAM_MEMBER_NOT_FOUND, Level.FINE);
     }
-    if (userProjectRole.equals(AllowedProjectRoles.DATA_SCIENTIST) && !owner.getEmail().equals(email)) {
+
+    //Data scientist can only remove themselves
```
Comment (tkakantousis) [inline hopsworks-api/src/main/java/io/hops/hopsworks/api/project/ProjectMembersService.java]: @robzor92 `scientists`

File: hopsworks-ear/test/spec/projects_spec.rb (hunk 2 above)
Comment (robzor92) [reply hopsworks-ear/test/spec/projects_spec.rb]: yes adding one for that too

File: hopsworks-api/src/main/java/io/hops/hopsworks/api/project/ProjectMembersService.java (hunk 3 above)
Comment (robzor92) [reply hopsworks-api/src/main/java/io/hops/hopsworks/api/project/ProjectMembersService.java]: resolved

File: hopsworks-common/src/main/java/io/hops/hopsworks/common/jobs/JobScheduler.java (hunk 1 above)
Comment (robzor92) [reply hopsworks-common/src/main/java/io/hops/hopsworks/common/jobs/JobScheduler.java]: resolved

File: hopsworks-api/src/main/java/io/hops/hopsworks/api/project/ProjectMembersService.java (hunk 4 above)
Comment (robzor92) [reply hopsworks-api/src/main/java/io/hops/hopsworks/api/project/ProjectMembersService.java]: resolved

File: hopsworks-ear/test/spec/projects_spec.rb (hunk 2 above)
Comment (robzor92) [reply hopsworks-ear/test/spec/projects_spec.rb]: added and now running tests on jenkins

**Comment by robzor92** [comment]
@tkakantousis It can be reviewed again, tests pass.

**Pull request merged**